packages/api/src/command/hie/unlink-patient-from-organization.ts

```diff
@@ -0,0 +1,379 @@
+    if (!organization) continue;
+
+    const identifier = organization.identifier?.find(identifier => identifier.value === oid);
+
+    if (identifier) {
+      matchingDocumentRefs.push(document);
+    }
+  }
+
+  return matchingDocumentRefs;
+}
+
+function getS3FileNameFromDocument(document: DocumentReferenceWithId): string | undefined {
+  const s3Attachment = document.content?.find(content => content.attachment?.url)?.attachment;
+  const fileName = s3Attachment?.title;
+
+  if (!fileName) {
+    console.error(`No file name found in document ${document.id}`);
+    return undefined;
+  }
+
+  return fileName;
+}
+
+async function findAndRemoveConversionResultsFromS3(
+  fileName: string,
+  dryRun = false,
+  log: typeof console.log
+): Promise<void> {
+  const dryRunMsg = getDryRunPrefix(dryRun);
+  try {
+    const conversionResultFileName = buildDocumentNameForConversionResult(fileName);
+    const normalizedFileName = buildDocumentNameForNormalizedConversion(fileName);
+    const cleanFileName = buildDocumentNameForCleanConversion(fileName);
+    const fromConverterFileName = buildDocumentNameForFromConverter(fileName);
+    const preConversionFileName = buildDocumentNameForPreConversion(fileName);
+
+    const fileNames = [
+      conversionResultFileName,
+      normalizedFileName,
+      cleanFileName,
+      fromConverterFileName,
+      preConversionFileName,
+    ];
+
+    const partialConversionFileNames = await getPartialConversionFileNames(preConversionFileName);
+    const allFileNames = [...fileNames, ...partialConversionFileNames];
+
+    const existingFiles = await Promise.all(
+      allFileNames.map(async fileName => {
+        try {
+          const exists = await s3Utils.fileExists(s3ConversionResultBucketName, fileName);
+          return exists ? fileName : undefined;
+        } catch (err) {
+          return undefined;
+        }
+      })
+    );
+
+    const validFiles = existingFiles.filter(fileName => fileName !== undefined) as string[];
+
+    for (const fileName of validFiles) {
+      log(`${dryRunMsg}Deleting file ${fileName} from S3`);
+      if (!dryRun) {
+        await s3Utils.deleteFile({ bucket: s3ConversionResultBucketName, key: fileName });
+      }
+    }
+  } catch (error) {
+    log("Error removing conversion results from S3:", errorToString(error));
+    throw error;
+  }
+}
+
+async function getPartialConversionFileNames(preConversionFileName: string): Promise<string[]> {
+  const partialConversionFileNames: string[] = [];
+  let index = 0;
+  let partialExists = true;
+  while (partialExists) {
+    const partialFileName = buildDocumentNameForPartialConversions(preConversionFileName, index);
+    try {
+      partialExists = await s3Utils.fileExists(s3ConversionResultBucketName, partialFileName);
+      if (partialExists) {
+        partialConversionFileNames.push(partialFileName);
+      }
+    } catch (err) {
+      partialExists = false;
+    }
+    index++;
+  }
```

Comment:
Can we add a max iterations so we fail after, say 99,999 iterations? Just in case a bug w/ `fileExists` doesn't make the API to hung. Here and all similar loops, please.

Reply:
No need for this logic anymore